### ISSUE TITLE
Replacing MergeTree with Final syntax take too long to response 

### ISSUE THREAD

**Issue by phongbui81062** [ClickHouse/ClickHouse#72214]
### Question

Hi guys, I got an issue with replacing mergetree table. My table has more than 5 billions records and data is comming to that tables almost 1k records per second (using kafka connect to insert). The reason is when I use final syntax, Clickhouse took too long to return the data, comparing without using final synctax
```
select count() from event
1.8 seconds to respond 
```
```
select count() from event final
30 mins to respond
```
Can you guy suggest me a solution for this?

**Comment by antaljanosbenjamin** [comment]
Yes, but check the results. With `FINAL` you probably get a lower, actual correct number. It has to filter out duplicates. This means it has to scan all of your data. Without `FINAL` you just get how many rows do you have without filtering out duplicates.

None of the tables meant for full table scans in large volume. You should write queries that are restricted on the primary key, or at least partially, so ClickHouse can efficiently skip records.
